### ISSUE TITLE
To update defaults on various systems

### ISSUE THREAD

**Issue by dipietrantonio** [Pawseyops/maali#118]
I suggest to drop gcc < 7 as default compilers on all systems if possible (not sure about crays). Also, we may want to update the default cuda etc.

**Comment by ddeeptimahanti** [comment]
Agree- but gcc/4.8.5 is still the default on Zeus. MWA has special interest in using gcc/5.5.0 on Galaxy/other systems- some codes are to be ported

**Comment by marcodelapierre** [comment]
Thanks for bringing this to the table @Halolegend94 .

In general I would agree for Zeus. For Magnus, I'd double check with Mark O, not sure about any Cray constraint.
One question though: do we have a strong reason to go ahead with this in the current systems, rather than postponing to the next one? Surely some maali builds take for ever because of the multiple compilers, but ... does it happen often these days? Just asking in the interest of best usage of staff time.

@ddeeptimahanti , if gcc/5.5.0 is of systematic use in any system then we should keep it in the defaults. Otherwise if it sporadic we can take it off; it doesn't mean it is banned from maali; it just means that it needs to be explicitly requested in the cygnet.

Two more notes:
1) I would still build general purpose libraries (e.g. MPI, LAPACK, FFT, HDF5, ...) with all compilers, so they're ready for use for the "special" cases.
2) If we go ahead with this, we will need to check the cygnet files for applications that NEED to use an older compiler.

**Comment by dipietrantonio** [comment]
It happened to me while building cmake on zeus. By the way, we should remove old cpu architectures right?

**Comment by marcodelapierre** [comment]
I have just removed KNL from Zeus!
For Sandybridge and the GCC versions I'd recommend a team discussion to evaluate, see email.

**Comment by dipietrantonio** [comment]
On magnus,
```
cdipietrantonio@nid00010:/scratch/pawsey0001/cdipietrantonio> echo $MAALI_DEFAULT_CPU_TARGET 
sandybridge ivybridge haswell
```
Software does not get compiled for broadwell which is the target architecture of the compute nodes if I am not mistaken. In fact I have compiled boost/.72.0 with this default and  now I am having issues loading modules. Am I missing something here?

**Comment by marcodelapierre** [comment]
mmh this needs to be double-checked with staff with more service than me, but I am pretty sure there's no broadwell on Magnus/Galaxy:
Magnus: haswell on both login and compute
Galaxy: ivybridge / sandybridge depending on workq or gpuq (can't recall which is which)

So I reckon the issue possibly comes from a different source
